Update overview picture

mattiarighi opened the pull request:
Close #1150.

![schematic](https://user-images.githubusercontent.com/25367743/59257384-1e8a1800-8c36-11e9-8816-5f86d0e7791c.png)

ledm commented:
I'm a little confused what the arrows mean. In many cases, it means are used by, (ie, the new units are needed by the convert units preprocesor), but elsewhere they indicate the order of actions. Personally, I would keep the arrows to indicate  order, and use some other kind of line to indicate "this information is needed by this preprocessor". Perhaps just a line with no arrowhead?

In the same vein, would it be a good time to put arrows between the processors, showing the default order? 

Similarly, the input data should go into the top left preprocessor (currently variable definition) instead of into the middle of the preprocessor box. You may need to move the title to the centre or right hand side because of this.

In the list of preprocessors, it would be nice to differentiate between the optional preprocessors that are specified in the recipe, and the preprocessors that run every time. Also, several of the preprocessors that run every time are not shown. (Load, and save files, for instance). 

In addition, the Sim card shape indicates recipe or recipe setting, the rounded corners shows a custom algorithm, but the box shape represents both data and operations. Perhaps we could use a different shape to distinguish them?

Sorry to add a ton of work - but this diagram is essential and I think it should be as clear as possible!

ledm commented:
Also, I don't think there should be a line linking the user config and the developper config.

ledm commented:
Personally, I think its the legend that is confusing things. We need to separate colours and shapes.

- blue: data
- yellow: preprocessors
- purple: recipes
- green: configuration
- red: diagnostics

- square: a specific file
- sim card: preprocessor setting
- rounded corner: operation

I'm not sure why we have filled and empty shapes. Any ideas?

This way, the preprocessors should be yellow rounded corners, and the diagnostics should be red rounded corners.

mattiarighi commented:
> I'm a little confused what the arrows mean. In many cases, it means are used by, (ie, the new units are needed by the convert units preprocesor), but elsewhere they indicate the order of actions. Personally, I would keep the arrows to indicate order, and use some other kind of line to indicate "this information is needed by this preprocessor". Perhaps just a line with no arrowhead?

Good points, I changed that.

> In the same vein, would it be a good time to put arrows between the processors, showing the default order?

I was not sure about this, since the order can be changed. But maybe showing the default is not a bad idea.

> Similarly, the input data should go into the top left preprocessor (currently variable definition) instead of into the middle of the preprocessor box. You may need to move the title to the centre or right hand side because of this.

Done that together with the previous point.

> In the list of preprocessors, it would be nice to differentiate between the optional preprocessors that are specified in the recipe, and the preprocessors that run every time. Also, several of the preprocessors that run every time are not shown. (Load, and save files, for instance).

Load and save are pretty trivial operations, and are already represented by the arrows.

> In addition, the Sim card shape indicates recipe or recipe setting, the rounded corners shows a custom algorithm, but the box shape represents both data and operations. Perhaps we could use a different shape to distinguish them?

We are using different colors for them, as shown in the legend, I would not complicate this further with additional shapes.

> I don't think there should be a line linking the user config and the developper config.

Actually, these are arrows converging with input data into the core: the idea was to show that one needs the information from the config files to read the data. But I removed that anyway.

> We need to separate colours and shapes.

I am open to suggestions.

> I'm not sure why we have filled and empty shapes. Any ideas?

No particular meaning was intended, but we can changed that as well.

See below the revised version (I also improved fonts and other things).

![schematic](https://user-images.githubusercontent.com/25367743/59275516-bac71580-8c5c-11e9-829f-1d014fcd9bd9.png)

ledm commented:
That looks much better! 

I'd very much suggest adding `Load file as an iris cube` and `save file as a netcdf` (or something) to the start and end of the preprocessor chain, as these are important operations that everyone already understands. It would form a good starting point to help people really anchor the idea of what a preprocessor operation is. 

Minor comments: 
* Horizontal regridding needs to be joined to the target grid scheme with a line.
* I think "diagnostic and graphical scripts" could be shortened to "Diagnostics", but perhaps add ".py, .R, or .ncl  script" in small font below.
* The red diagnostics colour are not in the legend.
* Do we want to add settings.yml & metadata.yml to the pre-processed input data? Alternatively, these could be a green sim-card box next to pre-processed input data, and feed into the diagnostics box. 


My suggestions for the shapes is:

Colours represent the different domains of ESMValTool

* blue: data
* yellow: preprocessor
* purple: recipe
* green: configuration
* red: diagnostic

And the shapes represent the type of object
* square: a specific file. (The config files would all need to be boxes now)
* rounded corner: an operation (this would require changing the yellow preprocessor box into a yellow rounded corner box, and the diagnostics to a red rounded corner box)
* sim card: a specific setting.

mattiarighi commented:
> I'd very much suggest adding Load file as an iris cube and save file as a netcdf (or something) to the start and end of the preprocessor chain, as these are important operations that everyone already understands. It would form a good starting point to help people really anchor the idea of what a preprocessor operation is.

I added this as an extra yellow box.

> Horizontal regridding needs to be joined to the target grid scheme with a line.

Fixed.

> I think "diagnostic and graphical scripts" could be shortened to "Diagnostics", but perhaps add ".py, .R, or .ncl script" in small font below.

Good suggestion, thanks!

> The red diagnostics colour are not in the legend.

Added.

> Do we want to add settings.yml & metadata.yml to the pre-processed input data? Alternatively, these could be a green sim-card box next to pre-processed input data, and feed into the diagnostics box.

These are actually temporary files generated on-the-run, I would not show them, also to avoid making the overview too crowded.

I also changed colors and shapes as you propose, please have a look...

![schematic2](https://user-images.githubusercontent.com/25367743/59283686-3d56d180-8c6b-11e9-882d-2bfee3778538.png)

mattiarighi commented:
Thanks for your feedback!
